web-api/src/personalization/rerank.rs

```diff
@@ -15,47 +15,46 @@
 use std::{cmp::Ordering, collections::HashMap, hash::Hash};
 
 use chrono::{DateTime, Utc};
-use itertools::{izip, Itertools};
-use xayn_ai_coi::{nan_safe_f32_cmp_desc, CoiSystem, UserInterests};
+use itertools::Itertools;
+use xayn_ai_coi::{nan_safe_f32_cmp, nan_safe_f32_cmp_desc, CoiSystem, UserInterests};
 
 use crate::models::{DocumentId, DocumentTag, PersonalizedDocument};
 
-#[derive(Copy, Clone, PartialEq, Eq, PartialOrd, Ord)]
-#[repr(transparent)]
-pub(super) struct Rank(usize);
```

Comment:
this wrapper doesn't work well with the elasticsearch scores, since we would have to resort those to get an index just to get the score again right afterwards, and the elasticsearch score is also computed differently than the interest and tag scores. hence i just moved the `to_score()` logic into `rank_keys_by_score()`. the `merge_as_score()` logic is extended to more than two inputs and moved into `rerank_by_scores()`.